dotCMS/src/main/java/com/dotcms/storage/FileSystemStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,346 @@
+            final String path,
+            final File file,
+            final Map<String, Object> extraMeta) {
+
+        if (!this.existsGroup(groupName)) {
+
+            throw new IllegalArgumentException("The groupName: " + groupName +
+                    ", does not have any file mapped");
+        }
+
+        final File groupFile = this.groups.get(groupName);
+
+        if (null != file && file.exists() && file.canRead() && groupFile.canWrite()) {
+
+            try {
+
+                final File destBucketFile = new File(groupFile, path);
+                FileUtils.copyFile(file, destBucketFile);
+            } catch (IOException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotRuntimeException(e);
+            }
+        } else {
+
+            throw new IllegalArgumentException("The file: " + file +
+                    ", is null, does not exist can not be read or bucket: " + groupName +
+                    " could not be written");
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object pushObject(final String groupName, final String path,
+            final ObjectWriterDelegate writerDelegate,
+            final Serializable object, final Map<String, Object> extraMeta) {
+
+        if (!this.existsGroup(groupName)) {
+
+            throw new IllegalArgumentException("The bucketName: " + groupName +
+                    ", does not have any file mapped");
+        }
+
+        final File groupFile = this.groups.get(groupName);
+
+        if (groupFile.canWrite()) {
+
+            try {
+
+                final File destBucketFile = new File(groupFile, path);
+                final String compressor = Config
+                        .getStringProperty("CONTENT_METADATA_COMPRESSOR", "none");
+                this.prepareParent(destBucketFile);
+
+                try (OutputStream outputStream = FileUtil
+                        .createOutputStream(destBucketFile.toPath(), compressor)) {
+
+                    writerDelegate.write(outputStream, object);
+                    outputStream.flush();
+                }
+            } catch (IOException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotRuntimeException(e);
+            }
+        } else {
+
+            throw new IllegalArgumentException("The bucket: " + groupName + " could not write");
+        }
+
+        return true;
+    }
+
+    private void prepareParent(final File file) {
+
+        if (!file.getParentFile().exists()) {
+
+            file.getParentFile().mkdirs();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Future<Object> pushFileAsync(final String groupName, final String path,
+            final File file, final Map<String, Object> extraMeta) {
+        return DotConcurrentFactory.getInstance().getSubmitter("StoragePool").submit(
```

Comment:
Codacy found an issue: [The String literal "StoragePool" appears 4 times in this file; the first occurrence is on line 210](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)